spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -81,13 +81,17 @@
 public class SparkCatalog extends BaseCatalog {
   private static final Set<String> DEFAULT_NS_KEYS = ImmutableSet.of(TableCatalog.PROP_OWNER);
 
+  public static final String PURGE_DATA_AND_METADATA = "purge-data-and-metadata";
```

Comment:
If we plan to introduce such a configure key,  it should be applied to all the catalogs (such as hive,hadoop and other custom catalogs)  , instead of a specific property for spark catalog ?

Reply:
Actually will use value of this property in Hive and Hadoop catalog, please take a look [on this method in Hive catalo, which will be called from spark catalog](https://github.com/apache/iceberg/blob/7779953bd2dad7990725391f9d5d8be077a7048d/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L148)

Reply:
I don't think `SparkCatalog` is the right place to set an Iceberg catalog property - i.e. what about Trino and Hive compute engines? Would they have a different behavior or not have the property at all?

I think if we wanted to add such a property, it should follow the same pattern as other catalog properties specifying it at the catalog level (i.e. a property in `CatalogProperties` and read via the properties passed to the catalog via `initialize` method.

Reply:
> I don't think `SparkCatalog` is the right place to set an Iceberg catalog property - i.e. what about Trino and Hive compute engines? Would they have a different behavior or not have the property at all?

Not sure that I understand why `SparkCatalog` is not a right place, please take a look on current implementation of [dropTable in SparkCatalog](https://github.com/apache/iceberg/blob/master/spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L237).  As you can see we are calling [dropTable method from Catalog.java ](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L284) where we always pass `true` for purge flag. So I just added ability to make it configurable via property passed to a catalog.
Currently Trino not dropping any data and metadata, just calls drop table from hive metastore